dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java

```diff
@@ -0,0 +1,1313 @@
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.baomidou.mybatisplus.core.metadata.IPage;
+import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+/**
+ * resources service impl
+ */
+@Service
+public class ResourcesServiceImpl extends BaseService implements ResourcesService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ResourcesServiceImpl.class);
+
+    @Autowired
+    private ResourceMapper resourcesMapper;
+
+    @Autowired
+    private UdfFuncMapper udfFunctionMapper;
+
+    @Autowired
+    private TenantMapper tenantMapper;
+
+    @Autowired
+    private UserMapper userMapper;
+
+    @Autowired
+    private ResourceUserMapper resourceUserMapper;
+
+    @Autowired
+    private ProcessDefinitionMapper processDefinitionMapper;
+
+    /**
+     * create directory
+     *
+     * @param loginUser login user
+     * @param name alias
+     * @param description description
+     * @param type type
+     * @param pid parent id
+     * @param currentDir current directory
+     * @return create directory result
+     */
+    @Transactional(rollbackFor = Exception.class)
+    public Result<Object> createDirectory(User loginUser,
+                                          String name,
+                                          String description,
+                                          ResourceType type,
+                                          int pid,
+                                          String currentDir) {
+        Result<Object> result = checkResourceUploadStartupState();
+        if (!result.getCode().equals(Status.SUCCESS.getCode())) {
+            return result;
+        }
+        String fullName = currentDir.equals("/") ? String.format("%s%s",currentDir,name) : String.format("%s/%s",currentDir,name);
+        result = verifyResource(loginUser, type, fullName, pid);
+        if (!result.getCode().equals(Status.SUCCESS.getCode())) {
+            return result;
+        }
+
+        if (checkResourceExists(fullName, 0, type.ordinal())) {
+            logger.error("resource directory {} has exist, can't recreate", fullName);
+            putMsg(result, Status.RESOURCE_EXIST);
+            return result;
+        }
+
+        Date now = new Date();
+
+        Resource resource = new Resource(pid,name,fullName,true,description,name,loginUser.getId(),type,0,now,now);
+
+        try {
+            resourcesMapper.insert(resource);
+            putMsg(result, Status.SUCCESS);
+            Map<Object, Object> dataMap = new BeanMap(resource);
```

Comment:
Recommended to use org.apache.commons.beanutils.BeanMap;

Reply:
![image](https://user-images.githubusercontent.com/4902714/108358825-26cb9080-722a-11eb-9832-494cff1f785b.png)
`org.apache.commons.beanutils.BeanMap` has been used in the latest code.